### DISCUSSION TITLE
Nach Update "JSON-Datei ist falsch formatiert."

### DISCUSSION THREAD

**Discussion by mnd182** [hoylabs/OpenDTU-OnBattery#1414] [Q&A] [answered]
Hallo liebe Community,

habe heute das Update auf 2024.11.20 ausgeführt ( ESP32 4mb ) - leider kommt nun immer die Fehlermeldung JSON-Datei ist falsch formatiert wenn ich mein Pin mapping einspielen möchte und ich weiß nun wirklich nicht mehr weiter. ( backup von alter version )

```json
[
    {
        "name": "DTUCMT5",
        "nrf24": {
            "miso": -1,
            "mosi": -1,
            "clk": -1,
            "irq": -1,
            "en": -1,
            "cs": -1
        },
        "eth": {
            "enabled": false
        },
        "cmt": {
            "clk": 14,
            "sdio": 12,
            "cs": 15,
            "fcs": 26,
            "gpio2": -1,
            "gpio3": -1
        },
        "display": {
            "type": 2,
            "data": 21,
            "clk": 22
		},
        "battery": {
            "rx": 17,
            "tx": -1
        },
        "victron": {
            "rx": 16,
            "tx": -1,
            "rx2": -1,
            "tx2": -1,
        }
    }
]
```

wahrscheinlich nur eine Kleinigkeit aber ich bin nicht soweit im Thema drin das ich hier irgendeinen Fehler erkennen kann - hab auch schon den freund google befragt.

**Comment by Solarteur** [comment, marked as answer]
Ich bin alles andere als ein Spezialist für pin mapping-Dateien, würde das Komma in der Zeile "tx2": -1, löschen, die Datei speichern und dann einen neuen Versuch starten (Vergleich mit anderen pin mapping-Dateien).

**Comment by mnd182** [comment]
Hallo Solarteur,

Fehler gefunden und behoben - Vielen Dank dafür :)
